### PR TITLE
rec: add Lua config to YAML settings

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

This  large PR (59 files changed at time of writing) allows settings originally defined by Lua config to be incorporated into the YAML settings introduced in 5.0.

One of the reasons this is big:  I had to tear apart the reading (from either Lua or YAML) of the settings and the activating of the settings. Originally those two things were intertwined. Now both method produce a single data structure (still named `LuaConfigItems`) that is fed to a function that activates the settings.

Included `rec_control show-yaml` and `reload-lua-config` support (for the YAML items that originated in Lua, we might add other reloadable settings later.). You can also dump the read config with `pdns_recursor --config`.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [X] included documentation (including possible behaviour changes)
- [X] documented the code
- [X] added or modified regression test(s)
- [X] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/settings/cxxsupport.cc [hunk 1]
```diff
@@ -682,3 +709,681 @@ std::string pdns::settings::rec::defaultsToYaml()
...
+  pbServer.logMappedFrom = pbConfig.logMappedFrom;
+}
+
+void fromLuaToRust(const FrameStreamExportConfig& fsc, pdns::rust::settings::rec::DNSTapFrameStreamServer& dnstap)
+{
+  for (const auto& server : fsc.servers) {
+    dnstap.servers.emplace_back(server);
+  }
+  dnstap.logQueries = fsc.logQueries;
+  dnstap.logResponses = fsc.logQueries;
```
Comment (rgacogne) [inline pdns/recursordist/settings/cxxsupport.cc:763]:
```suggestion
  dnstap.logResponses = fsc.logResponses;
```

**Review by rgacogne — CHANGES_REQUESTED** [review]
The code looks good and my tests with existing Lua configuration files have been successful. I noted a few small things but nothing big. Impressive work!

File: pdns/recursordist/settings/cxxsupport.cc [hunk 2]
```diff
@@ -682,3 +709,681 @@ std::string pdns::settings::rec::defaultsToYaml()
...
+  for (const auto& trustAnchor : dnssec.trustanchors) {
+    auto name = DNSName(std::string(trustAnchor.name));
+    for (const auto& dsRecord : trustAnchor.dsrecords) {
+      auto dsRecContent = std::dynamic_pointer_cast<DSRecordContent>(DSRecordContent::make(std::string(dsRecord)));
+      luaConfig.dsAnchors[name].emplace(*dsRecContent);
+    }
+  }
+  for (const auto& nta : dnssec.negative_trustanchors) {
+    luaConfig.negAnchors[DNSName(std::string(nta.name))] = std::string(nta.reason);
+  }
```
[the remark this answers is not in the document]
Comment (omoerbeek) [reply pdns/recursordist/settings/cxxsupport.cc:1078]: I'll double check this and either fix the code or add a comment.

File: pdns/recursordist/settings/cxxsupport.cc [hunk 3]
```diff
@@ -618,13 +623,35 @@ std::string pdns::settings::rec::defaultsToYaml()
...
     string val = arg().getDefault(var);
     if (pdns::settings::rec::oldKVToBridgeStruct(name, val, section, fieldname, type_name, rustvalue)) {
       map.emplace(std::pair{std::pair{section, fieldname}, pdns::rust::settings::rec::OldStyle{section, fieldname, name, std::move(type_name), std::move(rustvalue), false}});
     }
   }
+
+  // Should be generated
+  auto def = [&](const string& section, const string& name, const string& type) {
+    pdns::rust::settings::rec::Value rustvalue{};
+    rustvalue.u64_val = 24; // XXX
```
[the remark this answers is not in the document]
Comment (omoerbeek) [reply pdns/recursordist/settings/cxxsupport.cc:637]: Yes, this is a bit weird. I realized that (hence the XXX) while writing, but now I forgot. Will figure it out.

File: pdns/recursordist/settings/cxxsupport.cc [hunk 4]
```diff
@@ -634,7 +634,10 @@ std::string pdns::settings::rec::defaultsToYaml()
   // Should be generated
   auto def = [&](const string& section, const string& name, const string& type) {
     pdns::rust::settings::rec::Value rustvalue{};
-    rustvalue.u64_val = 24; // XXX
+    // Dirty hack: trustanchorfile_interval is the only u64 value, set the right default for it.
+    // All for all other values below, the default is either an empty string or an empty vector.
```
Comment (rgacogne) [inline pdns/recursordist/settings/cxxsupport.cc:638]:
```suggestion
    // And for all other values below, the default is either an empty string or an empty vector.
```

**Review by rgacogne — APPROVED** [review]

**Review by rgacogne — APPROVED** [review]

**Pull request merged**